consoles/serial_screen.pm

```diff
@@ -152,6 +151,45 @@ sub normalise_pattern {
     return $pattern;
 }
 
+=head2 do_read
+
+  my $num_read = do_read($buffer [, max_size => 2048][,timeout => undef]);
+
+Attempts to read up to max_size bytes from the C<<$self->{socket_fd}>> into buffer.
+The method returns as soon as some data is available, even if the given size has not been reached.
+Returns number of bytes read or undef on timeout. Note that 0 is a valid return code.
+If an failure occur the method croak.
+
+A undefined timeout will wait infinitive. A timeout of 0 just try to read once.
+=cut
+sub do_read
+{
+    my ($self, undef, %args) = @_;
```

Comment:
why you need this `undef` ?

Reply:
The second parameter is used in "call by reference" mode. Like other implementations [sysread](https://perldoc.perl.org/functions/sysread.html) or [Net::SSH2::Channel::read](https://metacpan.org/pod/Net::SSH2::Channel#read-(-buffer,-max_size-%5B,-ext-%5D-)) do.
In the line 189 it looks like ` $_[1] = $buffer;` which does the trick. 
And I thought this `undef` makes the reader aware of, that here is something special with the second parameter.